File: java/organizing-db-adapter/src/main/java/com/hexagonalchess/organizing/InMemoryInvitationRepository.java
package com.hexagonalchess.organizing;

import java.util.ArrayList;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;
import java.util.Optional;

public class InMemoryInvitationRepository implements InvitationRepository {

    private final Map<InvitationId, GameInvitation> store = new LinkedHashMap<>();

    @Override
    public void save(GameInvitation invitation) {
        store.put(invitation.id(), invitation);
    }

    @Override
    public Optional<GameInvitation> findById(InvitationId id) {
        return Optional.ofNullable(store.get(id));
    }

    @Override
    public List<GameInvitation> findByInvitingUserId(String userId) {
        return store.values().stream()
            .filter(i -> i.invitingUserId().equals(userId))
            .toList();
    }

    @Override
    public List<GameInvitation> findByInvitedUserId(String userId) {
        return store.values().stream()
            .filter(i -> i.invitedUserId().equals(userId))
            .toList();
    }
}

File: python/organizing_db_adapter/in_memory_invitation_repository.py
from organizing_policy.game_invitation import GameInvitation
from organizing_policy.invitation_id import InvitationId
from organizing_policy.invitation_repository import InvitationRepository


class InMemoryInvitationRepository(InvitationRepository):
    def __init__(self):
        self._store: dict[str, GameInvitation] = {}

    def save(self, invitation: GameInvitation) -> None:
        self._store[invitation.id.value] = invitation

    def find_by_id(self, invitation_id: InvitationId) -> GameInvitation | None:
        return self._store.get(invitation_id.value)

    def find_by_inviting_user_id(self, user_id: str) -> list[GameInvitation]:
        return [i for i in self._store.values() if i.inviting_user_id == user_id]

    def find_by_invited_user_id(self, user_id: str) -> list[GameInvitation]:
        return [i for i in self._store.values() if i.invited_user_id == user_id]
